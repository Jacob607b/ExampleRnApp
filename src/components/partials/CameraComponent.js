import { View, Text } from "react-native";
import React from "react";
import { TouchableOpacity, StyleSheet, Dimensions } from "react-native"; // Import Dimensions
import { Button } from "react-native-paper";
import {
  PinchGestureHandler,
  GestureHandlerRootView,
} from "react-native-gesture-handler";
import { Camera } from "expo-camera";
import { Icon } from "react-native-elements";

const CameraComponent = ({
  zoom,
  setCam,
  closeCamera,
  takePicture,
  flash,
  type,
  handleFlash,
}) => {
  return (
    <View style={styles.container}>
      <View style={styles.cameraContainer}>
        <GestureHandlerRootView style={{ flex: 1 }}>
          <PinchGestureHandler
            onGestureEvent={(event) => zoom(event)}
            testID="pinch"
          >
            <Camera
              ref={(ref) => setCam(ref)}
              testID="CamComponent"
              type={type}
              flashMode={flash}
              zoom={zoom}
            >
              {/* FLASH ICON TOGGLES FLASH WHEN PRESSED */}
              <TouchableOpacity
                onPress={handleFlash}
                style={styles.flash}
                testID="flash"
              >
                <Text>
                  {flash ? (
                    <Icon name="flash-on" size={40} color="#fff" />
                  ) : (
                    <Icon name="flash-off" size={40} />
                  )}
                </Text>
              </TouchableOpacity>
              {/* Capture Button */}
              <View
                style={{
                  paddingTop: "100%",
                  flexDirection: "column",
                  justifyContent: "center",
                  alignItems: "center",
                  marginVertical: "50%",
                }}
              >
                <TouchableOpacity
                  onPress={takePicture}
                  testID="takePicture"
                  style={{ backgroundColor: "#fff", borderRadius: 50 }}
                >
                  <Text>
                    <Icon name="camera" size={40} />
                  </Text>
                </TouchableOpacity>
                {/* Close Camera Button */}
                <Text style={styles.margin}>
                  <TouchableOpacity testID="closeCamera" onPress={closeCamera}>
                    <Text style={{ color: "#fff", fontSize: 20 }}>Close Camera</Text>
                  </TouchableOpacity>
                </Text>
              </View>
            </Camera>
          </PinchGestureHandler>
        </GestureHandlerRootView>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    height: "100%"
  },
  cameraContainer: {
    flex: 1,
    flexDirection: "row",
  },

  captureButton: {
    position: "absolute",
    bottom: 20,
    alignSelf: "center",
    alignItems: "center",
    height: 50,
    width: 50,
    borderRadius: 50,
    justifyContent: "center",
  },
  closeCamera: {
    marginTop: 10,
    marginBottom: 10,
  },
  flash: {
    position: "absolute",
    top: 20,
    right: 20,
    zIndex: 1,
  },
  margin: {
    marginBottom: 10,
    marginTop: 20,
  },
});

export default CameraComponent;
