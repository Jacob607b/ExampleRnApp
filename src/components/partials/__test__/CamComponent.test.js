import React from "react";
import { fireEvent, render, screen } from "@testing-library/react-native";
import CameraComponent from "../CameraComponent";

describe("Cam Component", () => {
 
  test("Cam Component RTL", ()=>{
    const setCam = jest.fn()
    const zoom = jest.fn();
    render(<CameraComponent zoom={zoom} setCam={setCam}/>)
    expect(screen).toMatchSnapshot();
    const pinchToZoom = screen.getByTestId("pinch");
    fireEvent(pinchToZoom, "onGestureEvent", {
      nativeEvent: {
        pointerCount: 2, // Simulate two fingers for pinch
        eventType: "pinch", // Specify the pinch event type
        scale: 2, // Simulate zooming in by scaling the gesture
      },
    })
  })
});
