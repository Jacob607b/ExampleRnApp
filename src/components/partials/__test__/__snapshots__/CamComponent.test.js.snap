// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cam Component Cam Component RTL 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <ViewManagerAdapter_ExpoCamera
        collapsable={false}
        faceDetectorSettings={Object {}}
        focusDepth={0}
        handlerTag={1}
        handlerType="PinchGestureHandler"
        onCameraReady={[Function]}
        onFacesDetected={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onMountError={[Function]}
        onPictureSaved={[Function]}
        onResponsiveOrientationChanged={[Function]}
        testID="pinch"
        zoom={[MockFunction]}
      >
        <View
          accessibilityState={
            Object {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            Object {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "position": "absolute",
              "right": 20,
              "top": 20,
              "zIndex": 1,
            }
          }
          testID="flash"
        >
          <Text>
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text />
                </View>
              </View>
            </View>
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "column",
              "justifyContent": "center",
              "marginVertical": "50%",
              "paddingTop": "100%",
            }
          }
        >
          <View
            accessibilityState={
              Object {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              Object {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 50,
                "opacity": 1,
              }
            }
            testID="takePicture"
          >
            <Text>
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </Text>
          </View>
          <Text
            style={
              Object {
                "marginBottom": 10,
                "marginTop": 20,
              }
            }
          >
            <View
              accessibilityState={
                Object {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                Object {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="closeCamera"
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 20,
                  }
                }
              >
                Close Camera
              </Text>
            </View>
          </Text>
        </View>
      </ViewManagerAdapter_ExpoCamera>
    </View>
  </View>
</View>
`;
